### ISSUE TITLE
Improve the sqlite3 tutorial

### ISSUE THREAD

**Issue by erlend-aasland** [python/cpython#95271]
Improve the tutorial using Diataxis as a guideline: https://diataxis.fr/tutorials/

The current tutorial is quite good, but it could make use of some improvements. The first that comes to mind, is:

- [Provide a clear picture before we start the tutorial](https://diataxis.fr/tutorials/#provide-a-complete-picture-before-they-start)
- Some parts of the tutorial could make use [clearer steps](https://diataxis.fr/tutorials/#describe-concrete-steps-not-abstract-concepts).
    - #95522 
- Make the wording more to the point; [avoid options and alternatives](https://diataxis.fr/tutorials/#ignore-options-and-alternatives).
- [Offer only minimum necessary explanation](https://diataxis.fr/tutorials/#offer-only-minimum-necessary-explanation)

Suggesting to split this up into multiple PRs, for the reviewers convenience.

**Comment by erlend-aasland** [comment]
cc. @CAM-Gerlach #diataxis

**Comment by CAM-Gerlach** [comment]
> So, I'm not particularly fond of the stock exchange example; it works well, but it is ... maybe a little bit dull[2](https://github.com/python/cpython/issues/95271#user-content-fn-2-3c5a058d53578077b1b3470a5a52e149)?

Maybe spice it up with some cryptocurrency? That's bound to be...exciting...these days... /s

As to the example, I'm really not sure. While being a relative subject matter n00b has been helpful in reviewing the docs from a more beginner perspective, I don't really have the required background to suggest something in this specific question, sorry.
